src/main/java/org/ehcache/integrations/shiro/EhcacheShiroManager.java

```diff
@@ -45,6 +45,8 @@
   private String cacheManagerConfigFile = "classpath:/org/ehcache/integrations/shiro/ehcache.xml";
   private boolean cacheManagerImplicitlyCreated = false;
 
+  private XmlConfiguration cacheConfiguration = null;
```

Comment:
now that I think about it, other fields most likely need to be volatile as well for the same reason